File: 1310xorQueriesOfASubarray/xorQueriesOfASubarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        arr = *itertools.accumulate(arr, operator.xor),
        return [arr[j] ^ arr[i - 1] if i else arr[j] for i, j in queries]

File: 1310xorQueriesOfASubarray/xorQueriesOfASubarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        Arrays.parallelPrefix(arr, (a,b) -> a ^ b);
        return Arrays.stream(queries).mapToInt($ -> $[0] != 0 ? arr[$[0] - 1] ^ arr[$[1]] : arr[$[1]]).toArray();
    }
}
